src/node/run-style-dictionary.js

```diff
@@ -30,45 +32,96 @@ async function cleanPlatformOutputDirs() {
 }
 
 export async function rerunStyleDictionaryIfSourceChanged(file) {
-  const { source } = styleDictionaryInstance.options;
-  const sourceFiles = new Set();
+  const { platforms } = styleDictionaryInstance.options;
+  let outputFiles = [];
   await Promise.all(
-    source.map(async (src) => {
-      const matches = await asyncGlob(src, { nodir: true, fs });
-      matches.forEach((m) => {
-        sourceFiles.add(`/${m}`);
+    Object.entries(platforms).map(([key, platform]) => {
+      return new Promise(async (resolve) => {
+        console.log(`${platform.buildPath}**`);
+        const outFiles = await asyncGlob(`${platform.buildPath}**`, {
+          nodir: true,
+          fs,
+        });
+        outputFiles = [...outputFiles, ...outFiles];
+        resolve();
       });
     })
   );
+  const allFiles = await asyncGlob("**/*", { nodir: true, fs });
+  const inputFiles = allFiles.filter((file) => !outputFiles.includes(file));
 
   // Send to analytics that user ran style dictionary, with how many source files (default 3)
   // to get a feeling of how much they are testing out
   mixpanel.track("Run Dictionary", {
-    sourceFiles: sourceFiles.size,
+    sourceFiles: inputFiles.size,
     platforms: styleDictionaryInstance.platforms,
   });
 
-  const isSourceFile = Array.from(sourceFiles).includes(file);
-  const isConfigFile = configPaths.includes(file);
+  const isInputFile = inputFiles.includes(file.replace(/^\//, ""));
 
   // Only run style dictionary if the config our source tokens were changed
-  if (!isSourceFile && !isConfigFile) {
+  if (!isInputFile) {
     return;
   }
 
-  await runStyleDictionary();
-  const encoded = await encodeContents([...Array.from(sourceFiles)]);
+  const encoded = await encodeContents(inputFiles);
   window.location.href = `${window.location.origin}/#project=${encoded}`;
+  await runStyleDictionary();
```

Comment:
In case style dictionary causes a fatal error, we do want to encode the contents, otherwise user loses their progress due to the code failing here, which can be unsuspected and frustrating during local debugging especially.